packages/pages/src/components/analytics/analytics.test.tsx

```diff
@@ -0,0 +1,150 @@
+      return <button onClick={async () => await track("foo click")} />;
+    };
+
+    render(
+      <AnalyticsProvider templateData={baseProps} requireOptIn={false}>
+        <MyButton />
+      </AnalyticsProvider>
+    );
+
+    fireEvent.click(screen.getByRole("button"));
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const generatedUrlStr = global.fetch.mock.calls[1][0];
+    const generatedUrl = new URL(generatedUrlStr);
+    expect(generatedUrl.searchParams.get("eventType")).toBe("fooclick");
+  });
+  it("should track a click with scoping", () => {
+    const MyButton = () => {
+      const { track } = useAnalytics();
+      return <button onClick={async () => await track("foo click")} />;
+    };
+
+    render(
+      <AnalyticsProvider templateData={baseProps} requireOptIn={false}>
+        <AnalyticsScope name={"header"}>
+          <AnalyticsScope name={"menu"}>
+            <MyButton />
+          </AnalyticsScope>
+        </AnalyticsScope>
+      </AnalyticsProvider>
+    );
+
+    fireEvent.click(screen.getByRole("button"));
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const generatedUrlStr = global.fetch.mock.calls[1][0];
+    const generatedUrl = new URL(generatedUrlStr);
+    expect(generatedUrl.searchParams.get("eventType")).toBe(
+      "header_menu_fooclick"
+    );
+  });
+  it("should track a click with a conversion", async () => {
+    const expectedConversionData = { cid: "123456", cv: "10" };
+
+    const MyButton = () => {
+      const { track, enableConversionTracking } = useAnalytics();
+      enableConversionTracking();
+      return (
+        <button
+          onClick={async () => await track("foo click", expectedConversionData)}
+        />
+      );
+    };
+
+    render(
+      <AnalyticsProvider templateData={baseProps} requireOptIn={false}>
+        <MyButton />
+      </AnalyticsProvider>
+    );
+
+    const user = userEvent.setup();
+    await user.click(screen.getByRole("button"));
+
+    await waitFor(() => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const mockCalls = global.fetch.mock.calls;
+
+      const generatedClickUrlStr = mockCalls[1][0];
+      const generatedClickUrl = new URL(generatedClickUrlStr);
+      expect(generatedClickUrl.searchParams.get("_yfpc")).toBeTruthy();
+
+      const generatedConversionUrlStr = mockCalls[2][0];
+      const generatedConversionUrl = new URL(generatedConversionUrlStr);
+      expect(generatedConversionUrl.searchParams.get("_yfpc")).toBeTruthy();
+      expect(generatedConversionUrl.searchParams.get("cid")).toBe(
+        expectedConversionData.cid
+      );
+      expect(generatedConversionUrl.searchParams.get("cv")).toBe(
+        expectedConversionData.cv
+      );
+    });
+  });
+  // TODO: figure out the right way to test window.location logic to credit listings with y_source param
```

Comment:
This PR?

Reply:
I did a lot of research trying to figure out the correct way of doing this but I don't want to block on it. I'm pretty sure it's redundant bc it's already tested for in the analytics library itself. Just trying to be thorough.

I don't think it needs to be part of this PR.